hoomd/hpmc/compute.py

```diff
@@ -325,18 +430,37 @@ def betaP(self):  # noqa: N802 - allow function name
             In MPI parallel execution, `betaP` is available on rank 0 only.
             `betaP` is `None` on ranks >= 1.
         """
-        if not numpy.isnan(self.sdf).all():
+        if not (numpy.isnan(self.sdf_compression).all()
+                or numpy.isnan(self.sdf_expansion).all()):
```

Comment:
```suggestion
        if self.sdf_compression is not None and self.sdf_expansion is not None:
```
Better yet,
```
sdf_compression = self.sdf_compression;
sdf_expansion = self.sdf_expansion;
if self.sdf_compression is not None and self.sdf_expansion is not None:
```
and then use `sdf_*` below instead of `self.sdf_*` to avoid making unnecessary data copies.